Ka-Block.safariextension/filter.js

```diff
@@ -3,20 +3,35 @@ function escape(domain) {
   return domain.replace(/[.]/g, '\\.')
 }
 
+// Get trigger url for block list rule
+function ruleURL(rule) {
+  return rule.trigger['url-filter']
+}
+
 // Create filter predicate function for blocked domains.
-function filter(domains) {
-  const all = domains.map(escape).join('|')
+function filter(rules) {
+  const all = rules.map(ruleURL).map(escape).join('|')
   const re = new RegExp('(?:^|\\.)(?:' + all + ')$', 'i')
   return re.test.bind(re)
 }
 
-// Predicate returns true for blocked domains.
-const blocked = filter(window.domains)
-
 // Respond to approval requests from start script.
-safari.application.addEventListener('message', function(event) {
-  if (event.name === 'canLoad') {
-    const host = new URL(event.message).hostname
-    event.message = blocked(host) ? 'block' : 'allow'
-  }
-}, true)
+function installCanLoad(rules) {
+  const blocked = filter(rules)
+  safari.application.addEventListener('message', function(event) {
+    if (event.name === 'canLoad') {
+      const host = new URL(event.message).hostname
+      event.message = blocked(host) ? 'block' : 'allow'
+    }
+  }, true)
+}
+
+function init() {
+  const req = new XMLHttpRequest()
+  req.responseType = 'json'
+  req.onload = function() { installCanLoad(req.response) }
+  req.open('GET', 'blockerList.json', true)
+  req.send()
```

Comment:
There is `safari.extension.baseURI` that returns `safari-extension://com.negativecode.safari.ka-block-UYW4V22L7E/74d929a9/`, but there's no way to use it create a file object through which we can access the json file.

Loading it through XHR with an extension-relative URL looks like the preferred method.